### PR TITLE
feat(derived_code_mappings): Enable task with only one project

### PR DESCRIPTION
**Pull request by armenzg**
We have some perf issues that we're investigating, however, we can enable this for just a project with few issues.

This is based on the work from #40717

### CODE REVIEW

File: src/sentry/tasks/derive_code_mappings.py [hunk 1]
```diff
@@ -34,7 +34,11 @@ def derive_missing_codemappings(dry_run=False) -> None:
     """
     Queue tasks to process each organization to derive missing codemappings.
     """
-    organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+    organizations = Organization.objects.filter(
+        status=OrganizationStatus.ACTIVE,
+        # Temporary change in order to enable the task until we fix perf issues
+        name__in=["sentry-test", "Sentry", "sentry", "Foo_org"],
```
Comment (armenzg) [inline src/sentry/tasks/derive_code_mappings.py:40]: Two of these are for the test. The other two are to hit the Sentry org. I'm just uncertain if it's uppercase or lowercase.

File: src/sentry/utils/sdk.py [hunk 2]
```diff
@@ -115,6 +115,8 @@
     "sentry.tasks.weekly_reports.schedule_organizations": settings.SAMPLED_DEFAULT_RATE,
     "sentry.tasks.weekly_reports.prepare_organization_report": 0.1,
     "sentry.profiles.task.process_profile": 0.01,
+    "sentry.tasks.derive_code_mappings.process_organizations": settings.SAMPLED_DEFAULT_RATE,
+    "sentry.tasks.derive_code_mappings.derive_code_mappings": settings.SAMPLED_DEFAULT_RATE,
```
Comment (armenzg) [inline src/sentry/utils/sdk.py:119]: Last week we did not see any tracing. Let's see what happens this time.

File: src/sentry/tasks/derive_code_mappings.py (hunk 1 above)
Comment (snigdhas) [reply src/sentry/tasks/derive_code_mappings.py:40]: Can we use the feature flag for this?

Comment (armenzg) [reply src/sentry/tasks/derive_code_mappings.py:40]: We could but did not want to open a 2nd PR on getsentry just for this since it is temporary.

**Comment by snigdhas** [comment]
Makes sense to me - just to clarify, we aren't submitting this until we improve the query right?

**Comment by armenzg** [comment]
> Makes sense to me - just to clarify, we aren't submitting this until we improve the query right?

I prefer to do it now *but* if you would rather not I'm fine as well. I have a way to get *some* performance data by using `SENTRY_DSN`. I run the following (I remove the `filter()` additions from this PR). I have a GH integration set up locally:

```
sentry shell -c "from sentry.tasks.derive_code_mappings import derive_missing_codemappings; derive_missing_codemappings.delay(dry_run=True)"
```

You can see it [here](https://sentry.io/organizations/sentry-test/performance/armenzg-test:d17f3787998d48608b2be116d7e8ca67/?transaction=sentry.tasks.derive_code_mappings.derive_code_mappings) (NOTE: I hack the code locally to pretend there are repositories associated in my Github org):
image

You can see that you could potentially make query changes locally to see if things would improve. For instance, the trace above takes 8 seconds which is very slow since we're querying a local project with barely any events.

This is not a full diff of what I have locally:
```diff
diff --git a/src/sentry/integrations/github/client.py b/src/sentry/integrations/github/client.py
index 9b2927e1f4..75f285e401 100644
--- a/src/sentry/integrations/github/client.py
+++ b/src/sentry/integrations/github/client.py
@@ -72,6 +72,7 @@ class GitHubClientMixin(ApiClient):  # type: ignore
     # https://docs.github.com/en/rest/git/trees#get-a-tree
     def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
         tree = []
+        return tree
         try:
             contents: Dict[str, Any] = self.get(
                 f"/repos/{repo_full_name}/git/trees/{tree_sha}",
diff --git a/src/sentry/tasks/derive_code_mappings.py b/src/sentry/tasks/derive_code_mappings.py
index 1f9042c068..b1fb90240a 100644
--- a/src/sentry/tasks/derive_code_mappings.py
+++ b/src/sentry/tasks/derive_code_mappings.py
@@ -59,6 +55,9 @@ def derive_code_mappings(organization_id: int, dry_run=False) -> None:
     Derive code mappings for an organization and save the derived code mappings.
     """
     organization: Organization = Organization.objects.get(id=organization_id)
+    set_tag("organization.slug", organization.slug)
+    # When you look at the performance page the user is a default column
+    set_user({"username": organization.slug})
     project_stacktrace_paths = identify_stacktrace_paths(organization)
     if not project_stacktrace_paths:
         return
```

**Comment by armenzg** [comment]
@snigdhas with PR #40781 I would not feel the need for this one since we can trigger perf tracing from our local dev.

**Comment by armenzg** [comment]
Closing for now. Let me know if you would rather land this.

**Pull request closed without merging**